carts/models.py

```diff
@@ -0,0 +1,106 @@
+"""
+Models for cart app
+"""
+from decimal import Decimal
+
+from django.core.validators import MinValueValidator
+from django.db import models
+from django.utils.translation import gettext_lazy as _
+
+from carts.constants import CART_STATUS, OPEN
+from products.models import AuditTimeStamp, Product
+from users.models import User
+
+
+class Cart(AuditTimeStamp):
+    """
+    Model class that represents Cart
+    """
+
+    status = models.CharField(max_length=40, choices=CART_STATUS, default=OPEN)
+    user = models.ForeignKey(User, on_delete=models.PROTECT, related_name="cart")
+    total_bill = models.DecimalField(
+        _("Total Payable"),
+        max_digits=20,
+        validators=(MinValueValidator(1),),
+        decimal_places=2,
+        default=Decimal(0),
+    )
+
+    def update_bill(self):
+        """
+        Updates the bill by traversing all the cart items
+        Returns:
+            None
+        """
+        # pylint: disable=no-member
+        total = Decimal(0)
+        for item in self.cart_items.all():
+            total += item.get_total()
+        self.total_bill = Decimal(total)
+        self.save()
+
+    def __str__(self):
+        return f"{self.user.name}'s Cart"
+
+
+class CartItem(AuditTimeStamp):
+    """
+    Model class that represents a CartItem
+    """
+
+    product = models.ForeignKey(Product, on_delete=models.CASCADE, related_name="+")
+    cart = models.ForeignKey(Cart, on_delete=models.CASCADE, related_name="cart_items")
+    quantity = models.PositiveIntegerField(
+        _("Product quantity to buy"), default=0, validators=(MinValueValidator(1),)
+    )
+    tax = Decimal(16 / 100)
```

Comment:
Remove the tax field for now please. Let's just deal with basic values

Reply:
Done